deploy/deploy-operator-go/main.go

```diff
@@ -15,248 +16,285 @@ import (
 	rbacv1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/rbac/v1"
 	"github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/yaml"
 	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
 )
 
 func main() {
 	pulumi.Run(func(ctx *pulumi.Context) error {
-		// Download file
-		filePath, cleanup, err := downloadFile("https://raw.githubusercontent.com/pulumi/pulumi-kubernetes-operator/v1.8.0/deploy/crds/pulumi.com_stacks.yaml")
-		if err != nil {
-			return err
+
+		var namespace = "default"
+		if ns := config.Get(ctx, "namespace"); ns != "" {
+			namespace = ns
 		}
-		crds, err := yaml.NewConfigFile(ctx, "crds", &yaml.ConfigFileArgs{
-			File: filePath,
-		})
-		if err != nil {
+		var namespaces []string
+		if err := config.GetObject(ctx, "namespaces", &namespaces); err != nil {
 			return err
 		}
+		if len(namespaces) == 0 {
+			namespaces = []string{namespace}
+		}
 
-		defer cleanup()
+		excludeCrds := config.GetBool(ctx, "include-crds")
 
-		operatorServiceAccount, err := corev1.NewServiceAccount(ctx, "operator-service-account", &corev1.ServiceAccountArgs{})
-		if err != nil {
-			return err
+		var deploymentOpts []pulumi.ResourceOption
+		if excludeCrds {
+			// Download file
+			filePath, cleanup, err := downloadFile("https://raw.githubusercontent.com/pulumi/pulumi-kubernetes-operator/v1.8.0/deploy/crds/pulumi.com_stacks.yaml")
```

Comment:
You should be able to pass the URL directly to the `File` argument of `yaml.ConfigFile`. (Same for all SDKs). I'm not sure if that's documented behavior or not.

Reply:
Ack, I've even seen that done elsewhere 😊 https://www.pulumi.com/registry/packages/kubernetes/api-docs/yaml/configfile/#inputs "Path or URL that uniquely identifies a file" ("uniquely identifies" -- odd way to put it 🤔)